wallets/core/src/wallet.ts

```diff
@@ -73,10 +73,16 @@ class Wallet<InstanceType = any> {
         network,
       });
     }
-    return await this.connect(network);
+    return await this.connect({ network });
   }
 
-  async connect(network?: Network, namespaces?: Namespace[]) {
+  async connect(params: {
```

Comment:
There is no need to change the interface here, you can consider it as: 
```
async connect(network, namespaces, extra: unknown)
```

I guess using `unknown` would be better here since it can be typed per provider separately which means `extra` can be typed inside `provider-ledger` itself andfor other providers they don't see anything regarding derivation path in `extra` which is not related to them.

And also please keep in mind, changing library's interface is not a good practice so we should carefully do that if there is no alternative solution.

Reply:
Considering that Trezor Wallet should use derivation path too, I think using 'unknown' is not a good approach here. To keep the library's interface as it is, I added a typed extra parameter.